documentation/jetty/modules/code/examples/src/main/java/org/eclipse/jetty/docs/programming/ArchitectureDocs.java

```diff
@@ -75,4 +79,78 @@ public void virtualVirtualThreads()
         HttpClient httpClient = new HttpClient(new HttpClientTransportOverHTTP(clientConnector));
         // end::virtualVirtual[]
     }
+
+    public static class EitherTask implements Invocable.Task
+    {
+        private final TryExecutor executor;
+        private final Deque<Invocable.Task> subTasks = new ConcurrentLinkedDeque<>();
+
+        public EitherTask(TryExecutor executor)
+        {
+            this.executor = executor;
+        }
+
+        @Override
+        public InvocationType getInvocationType()
+        {
+            return InvocationType.EITHER;
+        }
+
+        public void offer(Task task)
+        {
+            subTasks.add(task);
+        }
+
+        @Override
+        public void run()
```

Comment:
Technically, the whole `run()` method could be replaced with the following much simpler implementation, if `tryExecute()` optimizations are omitted:

```java
for (Task subTask = subTasks.pollFirst(); subTask != null; subTask = subTasks.pollFirst())
{
    InvocationType invocationType = Invocable.getInvocationType(subTask);
    invocationType.runWithoutBlocking(subTask, executor);
}
```

so I'm wondering if the doc should specify this, and how many details about the `tryExecute` optimization should be talked through?

Reply:
In this implementation, the calling thread never blocks, so the task could be NON_BLOCKING.
For an EITHER task, we need a example that does block sometimes.

Reply:
@lorban how about this version?

Reply:
I've reverted to the previous version